doc/_index.md

```diff
@@ -1,13 +1,85 @@
 ---
 title: "API & Pipeline"
-date: 2022-03-04
-draft: true
-description: "KernelCI API and Modular Pipeline"
-weight: 4
+date: 2023-09-01
+description: "KernelCI API and Pipeline"
+weight: 2
 ---
 
-This section covers the new KernelCI API as well as the initial pipeline using
-it.
+## API Overview
+
+The KernelCI API is a server-side service which provides two main features: a
+database abstraction and a publisher / subscriber interface.  Another important
+concept is the fact that users own the data they send to the API.  Let's have a
+quick look at how this all fits together.
+
+### Database Abstraction
+
+All the data managed by KernelCI is stored in a MongoDB database using node
+objects.  These can contain data about any part of the testing hierarchy such
+as a kernel revision, a build, static test results, runtime functional tests,
+regressions etc.  Each node has a parent so they form a tree
+([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)).  There is
+typically one root node for each kernel revision with lots of child nodes
+containing all the test data that relates to it.
+
+Each node object also has a state which can be used when orchestrating the
+pipeline.  For example, a node will be in the Running state while awaiting some
+results.  There's also a result value, to tell whether the job passed or
```

Comment:
What is a "job" in this context? A generic task to be run, K8s job, LAVA job or something else?

Reply:
See other comment below.  I'll resolve this one as it's just another occurrence of the same issue.